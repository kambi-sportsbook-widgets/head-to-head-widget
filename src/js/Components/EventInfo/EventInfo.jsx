import React from 'react';
import { translationModule } from 'kambi-widget-core-library';
import EventInfoContainer from './EventInfoContainer/EventInfoContainer';
import ScoreIndicator from './ScoreIndicator/ScoreIndicator';
import styles from './EventInfo.scss';

const t = translationModule.getTranslation.bind(translationModule);

const EventInfo = ({ homeName, homeScore, awayName, awayScore, start }) => {
   const date = new Date(start);
   const formatedDate = date.getDate() + ' ' + t('month' + [date.getMonth()]) + ' ' + date.getFullYear();
   return (
      <div>
         <EventInfoContainer>
            <div className={styles.homeName + ' KambiWidget-card-support-text-color'}>{homeName}</div>
            <ScoreIndicator score={homeScore} />
            <ScoreIndicator score={awayScore} />
            <div className={styles.awayName + ' KambiWidget-card-support-text-color'}>{awayName}</div>
         </EventInfoContainer>
         { <div className={styles.eventName + ' KambiWidget-card-support-text-color'}>{formatedDate}</div> }
      </div>)
};

EventInfo.propTypes = {
   homeName: React.PropTypes.string.isRequired,
   homeScore: React.PropTypes.number.isRequired,
   awayName: React.PropTypes.string.isRequired,
   awayScore: React.PropTypes.number.isRequired,
   start: React.PropTypes.number.isRequired
};

export default EventInfo;
