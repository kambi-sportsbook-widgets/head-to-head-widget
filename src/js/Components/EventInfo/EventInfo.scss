@import '../../shared';

:local(.general) {

}

:local(.name) {
   font-size: 16px;
   display: block;
   text-align: right;
   margin-right: 17px;
}

:local(.awayName) {
   flex: 1;
   font-size: 16px;
   text-align: left;
   margin-left: 17px;
}

:local(.homeName) {
   flex: 1;
   font-size: 16px;
   text-align: right;
   margin-right: 17px;
}

:local(.eventName) {
   font-size: 12px;
   margin-top: 10px;
   text-align: center;
}
