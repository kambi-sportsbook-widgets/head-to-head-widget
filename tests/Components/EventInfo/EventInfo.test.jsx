/* eslint-env jest */
import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import EventInfo from '../../../src/js/Components/EventInfo/EventInfo';

let renderer;

describe('EventInfo.render()', () => {

   beforeEach(() => {
      renderer = new ReactShallowRenderer();
   });

   it('renders correctly with required props', () => {
      expect(renderer.render(
         <EventInfo
            homeName="Poland"
            homeScore={3}
            awayName="Sweden"
            awayScore={1}
            start={new Date('2017-05-05').getTime()}
         />
      )).toMatchSnapshot();
   });

});
