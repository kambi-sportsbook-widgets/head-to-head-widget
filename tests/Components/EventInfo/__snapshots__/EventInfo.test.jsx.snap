// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventInfo.render() renders correctly with required props 1`] = `
<div>
  <EventInfoContainer>
    <div
      className="homeName KambiWidget-card-support-text-color"
    >
      Poland
    </div>
    <ScoreIndicator
      score={3}
    />
    <ScoreIndicator
      score={1}
    />
    <div
      className="awayName KambiWidget-card-support-text-color"
    >
      Sweden
    </div>
  </EventInfoContainer>
  <div
    className="eventName KambiWidget-card-support-text-color"
  >
    5 month4 2017
  </div>
</div>
`;
